src/main/java/com/adobe/cq/testing/junit/rules/ClassRuleUtils.java

```diff
@@ -28,10 +30,20 @@
 public class ClassRuleUtils {
     public static final Logger LOG = LoggerFactory.getLogger(ClassRuleUtils.class);
 
+    /**
+     * If index lanes should be dynamically detected
+     */
+    protected static final boolean DETECT_INDEXING_LANES = Boolean.getBoolean("cq.testing.detectIndexingLanes");
+
+    /**
+     * Indexing lanes to use if not detected dynamically
+     */
+    protected static final String[] DEFAULT_INDEXING_LANES = new String[]{"async", "fulltext-async"};
+
     // TODO way to make these props homogenous?
     public static final String LOGIN_TOKEN_AUTH = "it.logintokenauth";
 
-    public static class LoginTokenInstance extends ExistingInstance {
+    public static class LoginTokenInstance extends ConfigureIndexingLaneInstance {
```

Comment:
The chained inheritance doesn't seem like a good pattern here. What about we make a `ConfigurableInstance` class (in its own file 😄) that handles all the system props? Having the logic grouped in one place will help to handle all the combinations of flags.

Reply:
.... well, I can make `ConfigurableInstance` as a separate but afaict, `ClassRuleUtils#LoginTokenInstance` would still need to inherit from that. Similarly I guess `ClassRuleUtils#newInstanceRule` would just switch between `LoginTokenInstance` or `ConfigurableInstance`.

Am I missing something?

Reply:
I've moved `ConfiguredIndexLanesInstance` in a separate class at catholicon/aem-testing-clients@ebad5e5. We can, of course, discuss the naming (I don't want to drop `IndexLanes` from it - configured or configurable is probably up for discussion... since the lanes are hard-coded I would prefer "configured".

Reply:
I was thinking we can drop `LoginTokenInstance` and use only `ConfigurableInstance` as a replacement.

Reply:
Hmm... Yes, that maked sense. But, I think, that should be a separate issue. I can open one if you want.

I'm guessing then that the current flag for basic auth would be overloaded as use both basic auth and injected lanes. Or, maybe, you meant that we should have two different booleans and hence probably another `newInstanceRule`?